src/univariate/continuous/arcsine.jl

```diff
@@ -99,3 +99,13 @@ function gradlogpdf(d::Arcsine{T}, x::R) where {T, R <: Real}
     x == b && return TP(Inf)
     return TP(0.5 * (inv(b - x) - inv(x - a)))
 end
+function heslogpdf(d::Arcsine{T}, x::R) where {T, R <: Real}
+    TP = promote_type(T, R)
+    (a, b) = extrema(d)
+    # on the bounds, we consider the gradient limit inside the domain
+    # right side for the left bound,
+    # left side for the right bound
+    a < x <= b || return TP(-Inf)
+    x == b && return TP(Inf)
+    return TP(0.5 * (inv((x - a)^2) + inv((x - b)^2)))
+end
```

Comment:
It would be better to not base the return values on `T` and `R` explicitly. Instead one should perform as many calculations as possible and only return then `oftype(result, Inf)` (and similar things) in the cases where it is needed. This is currently not done for `gradlogpdf` it seems but since `heslogpdf` is a new addition it would be good to get it right from the start.

E.g., here one could write:
```suggestion
function heslogpdf(d::Arcsine, x::Real)
    a, b = extrema(d)
    result = (inv((x - a)^2) + inv((x - b)^2)) / 2
    # on the bounds, we consider the gradient limit inside the domain
    # right side for the left bound,
    # left side for the right bound
    return if x <= a
       oftype(result, -Inf)
    elseif x >= b
       oftype(result, Inf)
    else
       result
    end
end
```

Reply:
Thank you for your feedback, I'll adjust the code accordingly asap.